From 627c56fddad8e622830e9d8cfd2c4dd58ecf0921 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Tue, 30 Jun 2020 15:09:49 +0800
Subject: [PATCH] drm/rockchip: remove drm_atomic_set_property calls

drm_atomic_set_property isn't export function, so we set default prop
vale to instead of it.

Change-Id: I4acc6ddd045415aa180d467b45085609408e2447
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 36 ++++++++-------------
 1 file changed, 13 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index e962a89d3766..b4c199942387 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -1325,10 +1325,10 @@ static void rockchip_attach_connector_property(struct drm_device *drm)
 
 	drm_connector_list_iter_begin(drm, &conn_iter);
 	drm_for_each_connector_iter(connector, &conn_iter) {
-		ROCKCHIP_PROP_ATTACH(conf->tv_brightness_property, 100);
-		ROCKCHIP_PROP_ATTACH(conf->tv_contrast_property, 100);
-		ROCKCHIP_PROP_ATTACH(conf->tv_saturation_property, 100);
-		ROCKCHIP_PROP_ATTACH(conf->tv_hue_property, 100);
+		ROCKCHIP_PROP_ATTACH(conf->tv_brightness_property, 50);
+		ROCKCHIP_PROP_ATTACH(conf->tv_contrast_property, 50);
+		ROCKCHIP_PROP_ATTACH(conf->tv_saturation_property, 50);
+		ROCKCHIP_PROP_ATTACH(conf->tv_hue_property, 50);
 	}
 	drm_connector_list_iter_end(&conn_iter);
 #undef ROCKCHIP_PROP_ATTACH
@@ -1353,33 +1353,23 @@ static void rockchip_drm_set_property_default(struct drm_device *drm)
 	}
 	state->acquire_ctx = conf->acquire_ctx;
 
-#define CONNECTOR_SET_PROP(prop, val) \
-	do { \
-		ret = drm_atomic_set_property(state, &connector->base, \
-					      prop, \
-					      val); \
-		if (ret) \
-			DRM_ERROR("Connector[%d]: Failed to initial %s\n", \
-				  connector->base.id, #prop); \
-	} while (0)
-
 	drm_connector_list_iter_begin(drm, &conn_iter);
 	drm_for_each_connector_iter(connector, &conn_iter) {
 		struct drm_connector_state *connector_state;
 
 		connector_state = drm_atomic_get_connector_state(state,
 								 connector);
-		if (IS_ERR(connector_state))
-			DRM_ERROR("Connector[%d]: Failed to get state\n",
-				  connector->base.id);
-
-		CONNECTOR_SET_PROP(conf->tv_brightness_property, 50);
-		CONNECTOR_SET_PROP(conf->tv_contrast_property, 50);
-		CONNECTOR_SET_PROP(conf->tv_saturation_property, 50);
-		CONNECTOR_SET_PROP(conf->tv_hue_property, 50);
+		if (IS_ERR(connector_state)) {
+			DRM_ERROR("Connector[%d]: Failed to get state\n", connector->base.id);
+			continue;
+		}
+
+		connector_state->tv.brightness = 50;
+		connector_state->tv.contrast = 50;
+		connector_state->tv.saturation = 50;
+		connector_state->tv.hue = 50;
 	}
 	drm_connector_list_iter_end(&conn_iter);
-#undef CONNECTOR_SET_PROP
 
 	ret = drm_atomic_commit(state);
 	WARN_ON(ret == -EDEADLK);
-- 
2.35.3

